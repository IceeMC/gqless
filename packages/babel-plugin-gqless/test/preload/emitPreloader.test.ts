import { preload } from '../setup'

it('supports filtering', () => {
  const files = {
    a: `
      export default user => {
        user.a.b
        user.a.c
        user.b
      }
    `,
  }

  expect(preload(files)).toMatchSnapshot()
  expect(preload(files, '{a:{b}}')).toMatchSnapshot()
  expect(preload(files, '{a}')).toMatchSnapshot()
})
