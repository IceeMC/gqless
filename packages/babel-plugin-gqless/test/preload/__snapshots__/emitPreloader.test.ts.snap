// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`supports filtering 1`] = `() => {}`;

exports[`supports filtering 2`] = `
_user => {
  if (_user) {
    const _a = _user.a;

    if (_a) {
      _a.b;
    }
  }
}
`;

exports[`supports filtering 3`] = `
_user => {
  if (_user) {
    const _a = _user.a;

    if (_a) {
      _a.b;
      _a.c;
    }
  }
}
`;
